src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,265 @@
+
+        match event {
+            Event::Log(log) => Some(json!({ "log": log })),
+            Event::Metric(metric) => Some(json!({ "metric": metric })),
+            _ => panic!("The AppSignal sink does not support this event: {event:?}"),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl HttpSink for AppsignalSinkConfig {
+    type Input = serde_json::Value;
+    type Output = Vec<BoxedRawValue>;
+    type Encoder = AppsignalEventEncoder;
+
+    fn build_encoder(&self) -> Self::Encoder {
+        AppsignalEventEncoder {
+            transformer: self.encoding.clone(),
+        }
+    }
+
+    async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Bytes>> {
+        let uri = endpoint_uri(self.endpoint.clone(), "vector/events")?;
+        let request = Request::post(uri)
+            .header(
+                AUTHORIZATION,
+                format!("Bearer {}", self.push_api_key.inner()),
+            )
+            .header("Content-Encoding", "gzip");
+
+        let mut body = crate::serde::json::to_bytes(&events).unwrap().freeze();
+
+        let buffer = BytesMut::new();
+        let mut writer = GzEncoder::new(buffer.writer(), flate2::Compression::new(6));
+        writer.write_all(&body).expect("Writing to Vec can't fail");
+        body = writer
+            .finish()
+            .expect("Writing to Vec can't fail")
+            .into_inner()
+            .into();
+
+        request.body(body).map_err(Into::into)
+    }
+}
+
+pub(crate) async fn healthcheck(
+    uri: Uri,
+    push_api_key: String,
+    client: HttpClient,
+) -> crate::Result<()> {
+    let request = Request::get(uri).header(AUTHORIZATION, format!("Bearer {}", push_api_key));
+    let response = client.send(request.body(Body::empty()).unwrap()).await?;
+
+    match response.status() {
+        status if status.is_success() => Ok(()),
+        other => Err(super::HealthcheckError::UnexpectedStatus { status: other }.into()),
+    }
+}
+
+fn endpoint_uri(endpoint: String, path: &str) -> crate::Result<Uri> {
+    let mut uri = endpoint;
+    if !uri.ends_with('/') {
+        uri.push('/');
+    }
+    uri.push_str(path);
+    match uri.parse::<Uri>() {
+        Ok(u) => Ok(u),
+        Err(e) => Err(Box::new(BuildError::UriParseError { source: e })),
+    }
+}
+#[cfg(test)]
+mod test {
+    use futures::{future::ready, stream};
+    use serde::Deserialize;
+    use vector_core::event::{Event, LogEvent};
+
+    use crate::{
+        config::{GenerateConfig, SinkConfig, SinkContext},
+        test_util::{
+            components::{run_and_assert_sink_compliance, HTTP_SINK_TAGS},
+            http::{always_200_response, spawn_blackhole_http_server},
+        },
+    };
+
+    use super::{endpoint_uri, AppsignalSinkConfig};
+
+    #[test]
+    fn generate_config() {
+        crate::test_util::test_generate_config::<AppsignalSinkConfig>();
+    }
+
+    #[tokio::test]
+    async fn component_spec_compliance() {
```

Comment:
Thanks for adding unit tests that assert compliance!

Would it be possible to also add one that covers an error path? (using `assert_sink_error`)

Reply:
Ah I forgot I'd flagged error paths in both the int tests and unit.
It seems there isn't too much error path logic in this wrapper sink that is worth unit testing.
The assert sink error validation you added in the integration test looks sufficient to me, thanks for adding that.